projects/Mallard/src/helpers/settings/mutators.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * This exposes functions to update settings. Mutating a setting will update the
+ * value in Apollo's client cache, which will trigger all dependent React
+ * components to refresh. It also persists the new value in async storage.
+ */
+import { Settings, storeSetting, GdprSwitchSetting } from 'src/helpers/settings'
+import ApolloClient from 'apollo-client'
+
+const setSetting = (
+    name: keyof Settings,
+    client: ApolloClient<object>,
+    value: any,
+) => {
+    storeSetting(name, value as any)
+    client.writeData({ data: { [name]: value } })
+}
+
+const createMutator = <Name extends keyof Settings>(name: Name) => {
+    return setSetting.bind(undefined, name)
+}
+
+/**
+ * We never need to set up individual GDPR flags because they are managed by
+ * "buckets", so this is a shorthand to set any of the GDPR switch by its name.
+ */
+export const setGdprFlag = (
+    client: ApolloClient<object>,
+    name: keyof Settings,
+    value: GdprSwitchSetting,
+) => {
+    setSetting(name, client, value)
+}
+
+export const setIsWeatherShown = createMutator<'isWeatherShown'>(
```

Comment:
Yes, so that the the first argument of the mutator is strongly-typed. Sadly this means duplicating the name but in TS there's no way to use a "string type argument" as a value.

However if you see the `createMutator` declaration, you can see we enforce the strings to be the same (type and argument).

Reply:
As discussed IRL - you've written all the code to fix this and it _will_ work 💪

Reply:
Yup, will remove the explicit type parameters